packages/gatsby/src/redux/run-fast-filters.ts

```diff
@@ -1,45 +1,60 @@
-// @flow
-
-const { prepareRegex } = require(`../utils/prepare-regex`)
+import { IGatsbyNode } from "./types"
+import { GatsbyGraphQLType } from "../../index"
+import { prepareRegex } from "../utils/prepare-regex"
 const { makeRe } = require(`micromatch`)
 import { getValueAt } from "../utils/get-value-at"
 import _ from "lodash"
-const {
+import {
+  DbQuery,
+  IDbQueryQuery,
+  IDbQueryElemMatch,
   objectToDottedField,
   createDbQueriesFromObject,
   prefixResolvedFields,
-} = require(`../db/common/query`)
-const {
+} from "../db/common/query"
+import {
+  FilterOp,
+  FilterCacheKey,
+  FiltersCache,
+  FilterValueNullable,
   ensureEmptyFilterCache,
   ensureIndexByQuery,
   ensureIndexByElemMatch,
   getNodesFromCacheByValue,
   intersectNodesByCounter,
-} = require(`./nodes`)
+  IFilterCache,
+} from "./nodes"
+import { IGraphQLRunnerStats } from "../query/types"
+
+interface IInputQuery {
+  [key: string]: FilterValueNullable | IInputQuery
+}
+interface IPreparedQueryArg {
+  [key: string]: FilterValueNullable | IPreparedQueryArg
+}
 
 /**
  * Creates a key for one filterCache inside FiltersCache
- *
- * @param {Array<string>} typeNames
- * @param {DbQuery | null} filter If null the key will have empty path/op parts
- * @returns {FilterCacheKey} (a string: `types.join()/path.join()/operator` )
  */
-function createFilterCacheKey(typeNames, filter) {
+function createFilterCacheKey(
+  typeNames: Array<string>,
+  filter: DbQuery | null
+): FilterCacheKey {
   // Note: while `elemMatch` is a special case, in the key it's just `elemMatch`
   // (This function is future proof for elemMatch support, won't receive it yet)
   let f = filter
```

Comment:
I know this was there already, but I really don't like single char variable names. Letters are free!

Reply:
It was my doing anyways ;)
:shrug: I'll change it